### PR TITLE
Correct staff padding to include end padding

### PR DESCRIPTION
**Pull request by AaronDavidNewman**
Currently Formatter.formatToStave uses the pre-first-note padding but does not consider padding at the end of the measure.  Also, the formatting doesn't really consider end padding at all, so the actual end point of the bar is pretty inconsistent.

2 new metrics are defined for stave:  endPaddingMin and endPaddingMax.  default padding for calculation purposes is `padding + endPaddingMax`.  The formatter decides the current format fits within the measure if the right-most point is > `staffWidth - endPaddingMin` and < `staffWidth - endPaddingMax`.

Note that many test cases don't actually use formatToStave (even some that think they do) due to some other bugs that will be addressed in other changesets.

Here is a test case that _does_ use formatToStave.  Three images, pre-changes, changing only the padding, and changing the formatting to consider the padding.
Prior to this cs:
pr_formatter_before

Only considering additional end padding:
pr-correctpadding

Using both end padding and formatting to the end padding:
image

### CODE REVIEW

File: src/stave.ts [hunk 1]
```diff
@@ -62,6 +62,13 @@ export class Stave extends Element {
   protected bounds: Bounds;
   protected readonly modifiers: StaveModifier[];
 
+  // This is the sum of the padding that normally goes on left + right of a stave during
+  // drawing.  Used to size staves correctly with content width
+  static get defaultPadding() {
```
Comment (0xfe) [inline src/stave.ts:67]: Needs a return type: `static get defaultPadding() : number`.

File: src/formatter.ts [hunk 2]
```diff
@@ -693,10 +693,13 @@ export class Formatter {
       firstContext.getMetrics().totalLeftPx;
     let targetWidth = adjustedJustifyWidth;
     let actualWidth = shiftToIdealDistances(calculateIdealDistances(targetWidth));
-    const maxX = adjustedJustifyWidth + lastContext.getMetrics().notePx;
+    const musicFont = Flow.DEFAULT_FONT_STACK[0];
```
Comment (0xfe) [inline src/formatter.ts:696]: Not related to this PR, but lint error. Can you remove the `import { Vex } ...` from the top. (`Vex` is no longer used here.)

**Review by 0xfe — COMMENTED** [review]
Thanks. Minor comments.

**Comment by AaronDavidNewman** [comment]
OK!

**Comment by 0xfe** [comment]
LGTM, thanks

**Pull request merged**